### DISCUSSION TITLE
How to apply migrations required for a plugin?

### DISCUSSION THREAD

**Discussion by chrisdeely** [pretix/pretix#4313] [Q&A] [answered]
Hello! I'm new to python & django, so please forgive my ignorance of the processes here. 
I am attempting to write a plugin which will require adding a new table to the pretix database and a new model within my own plugin code. 

I have the migration file written, but how does pretix detect that it needs to be run? I've installed my plugin for development using `pip3 install -e .` and pretix detects it, allows it to be enabled and executes my code at the correct time.  However, the migration scripts are never run, so my code fails.

Hopefully I'm just missing something obvious.

**Comment by raphaelm** [comment]
You should be able to just run "python -m pretix migrate" :)

**Comment by chrisdeely** [comment]
Thanks for the quick response, @raphaelm. It certainly seems that would work for development purposes, but how does this happen when an end user decides to install a plug-in? Will I need to add the migrations to a Manifest or package them into the plug-in somehow?

**Reply by felixrindt** [reply, marked as answer]
Migrations are files (python modules) and part of the python package making up your Django app functioning as a pretix plugin. If you package the plugin, they will be part of your plugin package. Migrations will be executed no matter how your plugin package is installed when the "migrate" command is run.

Migrations are best created automatically using the "makemigrations" command, which places them in the correct directory in your project. It should work out of the box. You should not have to nor should be able to call your migration as a python script directly.

Maybe [the Django docs](https://docs.djangoproject.com/en/5.0/topics/migrations/) can help you pin down what's going on.
